compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/FuseAndHoistParallelLoops.cpp

```diff
@@ -151,6 +156,67 @@ struct FuseTilableDestinationProducers final : OpRewritePattern<scf::ForallOp> {
   }
 };
 
+struct FuseUnitLoopDestination final : OpRewritePattern<scf::ForallOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(scf::ForallOp forallOp,
+                                PatternRewriter &rewriter) const override {
+    if (!forallTripCountMatchesTarget(forallOp, 1)) {
+      return rewriter.notifyMatchFailure(forallOp,
+                                         "not a unit trip count loop");
+    }
+    DestinationStyleOpInterface dpsProducer;
```

Comment:
do we need to check for mapping attribute here?

Reply:
No need because if the producer is single use we want to do this fusion for any producer (at least that's my current assumption) and for any forall loop. If we find cases where that is not the case we can revisit accordingly.